ethers-providers/src/transports/http.rs

```diff
@@ -94,7 +104,24 @@ impl Provider {
     /// let provider = Http::new(url);
     /// ```
     pub fn new(url: impl Into<Url>) -> Self {
-        Self { id: AtomicU64::new(0), client: Client::new(), url: url.into() }
+        Self { id: AtomicU64::new(0), client: Client::new(), url: url.into(), authorization: None }
+    }
+
+    /// Initializes a new HTTP Client with authentication
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use ethers_providers::{Authorization, Http};
+    /// use url::Url;
+    ///
+    /// let url = Url::parse("http://localhost:8545").unwrap();
+    /// let provider = Http::new_with_auth(url, Authorization::Basic("admin".into(), "good_password".into()));
+    /// ```
+    pub fn new_with_auth(url: impl Into<Url>, auth: Authorization) -> Self {
+        let mut provider = Self::new(url);
+        provider.authorization = Some(auth);
+        provider
```

Comment:
same point,
instead of rolling our own `Auth` types, let's add an additional constructor that accepts a `reqwest::Client`, this way a user can configure any default headers

Reply:
however if we use the `Authorization` type for the ws auth, then this is fine.

still would like to see an additional function that accepts a `reqwest::Client`